src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

```diff
@@ -168,6 +168,8 @@ services:
       - "@prestashop.adapter.legacy.context"
       - "@prestashop.adapter.data_provider.warehouse"
       - "@prestashop.adapter.data_provider.carrier"
+      - "@=service('prestashop.adapter.legacy.configuration').get('PS_DIMENSION_UNIT')"
+      - "@=service('prestashop.adapter.legacy.configuration').get('PS_WEIGHT_UNIT')"
```

Comment:
I suggest to use configuration service as a param instead of config value, because : 

- if later we need a new configuration value, we will no longer need to change the constructor of the class.

Reply:
But then you inject a big object instead of a small property 😮 isn't it a waste of memory?

The [work in progress](https://github.com/PrestaShop/ADR/pull/25) ADR for BC breaks states that Symfony service don't have to preserve backward compatibility for their constructors so we will be able to modify the constructor without BC break blocker.

Reply:
@matks IMHO it's not only about BC Breaks. When's the limit where it's better to use the service instead of values? Four constructor arguments? Ten arguments? How many construct arguments is too many? :D I see using a service directly as a future-proof solution and having a lot of arguments smells like a constructor over-injection. **On the other hand** with that approach we don't need to worry about the configuration key name and if it does not produce a BC Break... =)

Reply:
> But then you inject a big object instead of a small property :open_mouth: isn't it a waste of memory?

@matks are you sure that `PrestaShop\PrestaShop\Adapter\Configuration::get` is "a small property" ?

Reply:
I am sure what will be returned is a small property, yes. I think you want to mean something, feel free to make it explicit 😄